src/restic/backend/s3/s3.go

```diff
@@ -97,7 +97,12 @@ func (be s3) Load(h backend.Handle, p []byte, off int64) (int, error) {
 	defer func() {
 		be.connChan <- struct{}{}
 	}()
-	return io.ReadFull(obj, p)
+
+	// This may not read the whole object, so ensure object
+	// is closed to avoid duplicate connections.
+	n, err := io.ReadFull(obj, p)
+	obj.Close()
```

Comment:
This discards the error that `Close()` may have returned. What do you think about returning the error from `Close()` in case `ReadFull()` did not return any error?